src/CardInfo.js

```diff
@@ -1,74 +1,59 @@
+            ...card,            
+            description
         }        
         
-        await onChangeCard(body, this.props);       
-
-        this.setState({            
-            isEditing: false
-        });
+        await onChangeCard(changeCard, card);        
     }
 
-    handleCancelButton = (event) => {
-        this.setState({
-            description: this.props.description,
-            isEditing: false
-        });
-    }
+    handleChangeStatus = async(event) => {
+        const {onChangeCard, card} = this.props;
 
-    handleChangeDescription = (event) => {
-        this.setState({
-            description: event.target.value
-        });        
+        let changeCard = {
+            ...card,
+            status: event.value            
+        }        
+        
+        await onChangeCard(changeCard, card);        
     }
 
     render() {
-        let {isEditing, description} = this.state;
-        let {changeDescription, status} = this.props;        
+        let {changeDescription, statuses} = this.props;  
+        let {status, description} = this.props.card;
+
+        let statusOptions = statuses && statuses.map(value => ({value, label: value}));
 
         return (
-            <div> 
-                <div>{`Status: ${status}`}</div>
-                <div class="flex-row">
-                    <div>
-                        {isEditing && changeDescription 
-                            ? <textarea onChange={this.handleChangeDescription}>{description}</textarea>
-                            : description
-                        }
-                    </div>
-                    {changeDescription && 
-                    <div>
-                        {isEditing 
-                            ? <Fragment>
-                                <button onClick={this.handleSaveButton}>Save</button> 
-                                <button onClick={this.handleCancelButton}>Cancel</button>
-                            </Fragment>
-                            : <button onClick={this.handleChangeButton}>Edit</button>
-                        }
-                    </div>}
-                </div>            
+            <div>
+                <div>
+                    {changeDescription                        
+                        ? <Fragment> 
+                            <Select
+                                defaultValue={{value: status, label: status}}
+                                options={statusOptions}                            
+                                onChange={this.handleChangeStatus}
+                            />
+                            <TextEditor
+                                text={description}
+                                onChangeText={this.handleChangeDescription}
+                            />
+                        </Fragment>
+                        : <div>
+                            <p>{`Status: ${status}`}</p>
+                            <p>{description}</p>
```

Comment:
render Select dynamically here and TextEditor (since not in edit mode it renders just text)